### PR TITLE
Add GetCurrentLowestXScore query function

### PR DESCRIPTION
**Pull request by relyt29**
## Purpose of Changes and their Description

This adds a query function that lets inferers, forecasters, and reputers to check what the cutoff boundary is for making it into the active set, before sending a transaction on chain

## Are these changes tested and documented?

Adds `TestGetCurrentLowestXScore` functions in `x/emissions/keeper/queryserver/query_server_score_test.go`

### CODE REVIEW

**Review by fernandofcampos — APPROVED** [review]

File: x/emissions/keeper/queryserver/query_server_score.go [hunk 1]
```diff
@@ -114,6 +115,46 @@ func (qs queryServer) GetWorkerInferenceScoresAtBlock(
 	return &types.QueryWorkerInferenceScoresAtBlockResponse{Scores: &workerInferenceScores}, nil
 }
 
+func (qs queryServer) GetCurrentLowestInfererScore(
```
Comment (RedBird96) [inline x/emissions/keeper/queryserver/query_server_score.go:118]: This is different with `socre_utils/GetLowScoreFromAllInferences()`?. I think we can use this function for this query, isn't it?

File: x/emissions/keeper/queryserver/query_server_score.go [hunk 2]
```diff
@@ -144,6 +185,46 @@ func (qs queryServer) GetWorkerForecastScoresAtBlock(
 	return &types.QueryWorkerForecastScoresAtBlockResponse{Scores: &workerForecastScores}, nil
 }
 
+func (qs queryServer) GetCurrentLowestForecasterScore(
```
Comment (RedBird96) [inline x/emissions/keeper/queryserver/query_server_score.go:187]: Same above with `GetLowScoreFromAllForecasts()`?

File: x/emissions/keeper/queryserver/query_server_score.go (hunk 1 above)
Comment (relyt29) [reply x/emissions/keeper/queryserver/query_server_score.go:118]: we might be able to let me check

Comment (relyt29) [reply x/emissions/keeper/queryserver/query_server_score.go:118]: yeah i think we can, will replace

**Comment by relyt29** [comment]
@RedBird96 can you take another look

**Review by RedBird96 — APPROVED** [review]
lgtm

**Pull request merged**